cmd-car-split.go

```diff
@@ -0,0 +1,333 @@
+			}
+
+			sl, err := writeSubsetNode(currentSubsetInfo, bufferedWriter)
+			if err != nil {
+				return fmt.Errorf("failed to write subset node: %w", err)
+			}
+			subsetLinks = append(subsetLinks, sl)
+
+			epochNode, err := qp.BuildMap(ipldbindcode.Prototypes.Epoch, -1, func(ma datamodel.MapAssembler) {
+				qp.MapEntry(ma, "kind", qp.Int(int64(iplddecoders.KindEpoch)))
+				qp.MapEntry(ma, "epoch", qp.Int(int64(epoch)))
+				qp.MapEntry(ma, "subsets",
+					qp.List(-1, func(la datamodel.ListAssembler) {
+						for _, sl := range subsetLinks {
+							qp.ListEntry(la, qp.Link(sl))
+						}
+					}),
+				)
+			})
+			if err != nil {
+				return fmt.Errorf("failed to construct epochNode: %w", err)
+			}
+
+			_, err = writeNode(epochNode, bufferedWriter)
+			if err != nil {
+				return fmt.Errorf("failed to write epochNode: %w", err)
+			}
+
+			return closeFile(bufferedWriter, currentFile)
+		},
+	}
+}
+
+func writeSubsetNode(currentSubsetInfo subsetInfo, bufferedWriter *bufio.Writer) (datamodel.Link, error) {
+	subsetNode, err := qp.BuildMap(ipldbindcode.Prototypes.Subset, -1, func(ma datamodel.MapAssembler) {
+		qp.MapEntry(ma, "kind", qp.Int(int64(iplddecoders.KindSubset)))
+		qp.MapEntry(ma, "first", qp.Int(int64(currentSubsetInfo.firstSlot)))
+		qp.MapEntry(ma, "last", qp.Int(int64(currentSubsetInfo.lastSlot)))
+		qp.MapEntry(ma, "blocks",
+			qp.List(-1, func(la datamodel.ListAssembler) {
+				for _, bl := range currentSubsetInfo.blockLinks {
+					qp.ListEntry(la, qp.Link(bl))
+				}
+			}))
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to write a subsetNode: %w", err)
+	}
+
+	cid, err := writeNode(subsetNode, bufferedWriter)
+	if err != nil {
+		return nil, fmt.Errorf("failed to write a subsetNode: %w", err)
+	}
+
+	return cidlink.Link{Cid: cid}, nil
+}
+
+func closeFile(bufferedWriter *bufio.Writer, currentFile *os.File) error {
+	err := bufferedWriter.Flush()
+	if err != nil {
+		return fmt.Errorf("failed to flush buffer: %w", err)
+	}
+
+	err = currentFile.Close()
+	if err != nil {
+		return fmt.Errorf("failed to close file: %w", err)
+	}
+	return nil
+}
+
+func writeNode(node datamodel.Node, w io.Writer) (cid.Cid, error) {
+	node = node.(schema.TypedNode).Representation()
+	var buf bytes.Buffer
+	err := dagcbor.Encode(node, &buf)
+	if err != nil {
+		return cid.Cid{}, err
+	}
+
+	data := buf.Bytes()
+
+	bd := cid.V1Builder{MhLength: -1, MhType: uint64(multicodec.Sha2_256), Codec: uint64(multicodec.DagCbor)}
+	cd, err := bd.Sum(data)
+	if err != nil {
+		return cid.Cid{}, err
+	}
+
+	c := []byte(cd.KeyString())
+
+	sizeVi := appendVarint(nil, uint64(len(c))+uint64(len(data)))
+
+	if _, err := w.Write(sizeVi); err == nil {
+		if _, err := w.Write(c); err == nil {
+			if _, err := w.Write(data); err != nil {
+				return cid.Cid{}, err
+			}
+
+		}
+	}
+	return cd, nil
+}
+
+func appendVarint(tgt []byte, v uint64) []byte {
+	for v > 127 {
+		tgt = append(tgt, byte(v|128))
+		v >>= 7
+	}
+	return append(tgt, byte(v))
+}
```

Comment:
This implementation doesn't seem to correspond to https://github.com/filecoin-project/go-leb128/blob/8d79a54895432c34a67d29f701fdf7180ded9047/leb128.go#L10-L23

Reply:
I copy pasted it from go-fil-dataprep:

https://github.com/anjor/go-fil-dataprep/blob/bf61394b8cd7ddb05fb9c652e8c55312ada118e1/cmd/data-prep/fil-data-prep/fil-data-prep.go#L169-L187

Reply:
Happy to switch it to leb128 if you prefer that.

Reply:
That was copied from 
https://github.com/anjor/anelace/blob/e7a70b075964d420afb565a96a2cb14f11771159/internal/util/encoding/util.go#L30-L36

which was originally written by RIba.

Reply:
Regardless I changed it to using leb128 @gagliardetto